core/src/main/java/org/apache/iceberg/BaseMetadataTableScan.java

```diff
@@ -31,11 +37,30 @@ protected BaseMetadataTableScan(TableOperations ops, Table table, Schema schema,
     super(ops, table, schema, context);
   }
 
+  /**
+   * @return if metadata table scan is for all snapshots, ie 'all_x' metadata tables
+   */
+  protected boolean allScan() {
```

Comment:
+1, I think `allScan` is not very descriptive

Reply:
In fact, what about removing this flag completely and overriding `planFiles` in `BaseAllMetadataTableScan`?

Reply:
The current implementation of `planFiles` seems a little bit weird with this flag.

Reply:
@aokolnychyi yea i had that version in the previous iteration, having a method like:

BaseMetadataTableScan.scanAllManifests(); //like planFiles but across all manifests.

and then having:

@Override
BaseAllMetadataTableScan.planFiles() {
   super.scanAllManifests();
}

@Override
AllDataFileTableScan.planFiles() {
  super.scanAllManifests();
}

Is that better than this?  I can revert back to that version.

Reply:
The reason I changed to this approach , it's slightly more clear in forcing implementators to have to override this flag, rather than have to find this method just by reading the code , but it's not a big deal either way as it's fairly internal.